sql/lease.go

```diff
@@ -151,11 +152,13 @@ func (s LeaseStore) Acquire(txn *client.Txn, tableID ID, minVersion DescriptorVe
 		}
 		return nil
 	})
```

Comment:
Ah, yes. That was my worry.. 

I tried to change `s.db.Txn` to always `error`, but that didn't work as some callers need to call `pErr.GetDetail`. We could strip some info (e.g., `pErr.txn`), but wasn't totally sure whether we should do.